### PR TITLE
Remove conhost telemetry

### PR DESCRIPTION
**Pull request by lhecker**
The `Telemetry` class was implemented as a singleton which stood in
my long-term goal to remove all global variables from the project.
Most telemetry captured by it hasn't been looked at for a long time
and just as much is now pointless (e.g.,`_fCtrlPgUpPgDnUsed`).
This removes the code.

## Validation Steps Performed
* Still compiles ✅

### CODE REVIEW

**Review by zadjii-msft — APPROVED** [review]

File: src/host/telemetry.hpp [hunk 1]
```diff
@@ -1,186 +1,17 @@
...
-    // Quick edit copy and paste usage telemetry
-    unsigned int _uiQuickEditCopyProcUsed;
-    unsigned int _uiQuickEditCopyRawUsed;
-    unsigned int _uiQuickEditPasteProcUsed;
-    unsigned int _uiQuickEditPasteRawUsed;
 };
-
-// Log the RIPMSG through telemetry, and also through a normal OutputDebugStringW call.
-// These are drop-in substitutes for the RIPMSG0-4 macros from /windows/Core/ntcon2/conhost/consrv.h
-#define RIPMSG0(flags, msg) Telemetry::Instance().LogRipMessage(msg);
```
Comment (DHowett) [inline src/host/telemetry.hpp:182]: YESS

File: src/host/srvinit.cpp [hunk 2]
```diff
@@ -428,11 +428,6 @@ HRESULT ConsoleCreateIoThread(_In_ HANDLE Server,
                                               [[maybe_unused]] PCONSOLE_API_MSG connectMessage)
 try
 {
-    // Create a telemetry instance here - this singleton is responsible for
-    // setting up the g_hConhostV2EventTraceProvider, which is otherwise not
```
Comment (DHowett) [inline src/host/srvinit.cpp:432]: who sets this up now?

File: src/server/ApiDispatchers.cpp [hunk 3]
```diff
@@ -781,14 +755,11 @@ static DWORD TraceGetThreadId(CONSOLE_API_MSG* const m)
...
-        Telemetry::Instance().LogApiCall(Telemetry::ApiCall::ReadConsoleOutputAttribute);
         break;
     case CONSOLE_ASCII:
-        Telemetry::Instance().LogApiCall(Telemetry::ApiCall::ReadConsoleOutputCharacter, false);
         break;
     case CONSOLE_REAL_UNICODE:
     case CONSOLE_FALSE_UNICODE:
-        Telemetry::Instance().LogApiCall(Telemetry::ApiCall::ReadConsoleOutputCharacter, true);
         break;
     }
```
Comment (DHowett) [inline src/server/ApiDispatchers.cpp:765]: wait somebody's gonna find this in the future and choke on their Soylent

```c++
switch (a->StringType)
{
	case 1:
		break;
	case 2:
		break;
	case 3:
		break;
}
```

complicated no-op. it keeps our enemies off our trail

File: src/server/ApiDispatchers.cpp [hunk 4]
```diff
@@ -428,14 +419,11 @@ static DWORD TraceGetThreadId(CONSOLE_API_MSG* const m)
     switch (a->ElementType)
     {
     case CONSOLE_ATTRIBUTE:
```
Comment (DHowett) [inline src/server/ApiDispatchers.cpp:421]: same as below

File: src/server/ApiDispatchers.cpp [hunk 5]
```diff
@@ -132,11 +127,9 @@ static DWORD TraceGetThreadId(CONSOLE_API_MSG* const m)
     const auto a = &m->u.consoleMsgL1.GetConsoleInput;
     if (WI_IsFlagSet(a->Flags, CONSOLE_READ_NOREMOVE))
     {
-        Telemetry::Instance().LogApiCall(Telemetry::ApiCall::PeekConsoleInput, a->Unicode);
     }
     else
     {
-        Telemetry::Instance().LogApiCall(Telemetry::ApiCall::ReadConsoleInput, a->Unicode);
     }
```
Comment (DHowett) [inline src/server/ApiDispatchers.cpp:133]: _rolling on the floor at this point_

**Review by DHowett — CHANGES_REQUESTED** [review]

File: src/host/srvinit.cpp (hunk 2 above)
Comment (lhecker) [reply src/host/srvinit.cpp:432]: `src/host/exe/exemain.cpp` here: https://github.com/microsoft/terminal/pull/16253/files#diff-47e1cca9863ea63bf26633f56e642bf72560cec16f786631cec4f3d9d885a073R225

**Review by DHowett — APPROVED** [review]

File: src/host/srvinit.cpp (hunk 2 above)
Comment (lhecker) [reply src/host/srvinit.cpp:432]: So... about that `TraceLoggingRegister` call...
mhm

Comment (DHowett) [reply src/host/srvinit.cpp:432]: _yup!_

How did you notice?

**Pull request merged**